node/src/components/small_network.rs

```diff
@@ -928,7 +928,7 @@ where
                             count,
                             exclude,
                         );
-                        responder.respond(sent_to).ignore()
+                        responder.into_inner().respond(Some(sent_to)).ignore()
```

Comment:
If we had `respond` defined on `AutoclosingResponder` we wouldn't need `into_inner()` or `Some(_)` and just pass in `responder.respond(_)`.

Reply:
I agree with this. Additionally, we could also have a `respond_none`, `respond_empty` or any better name for when we want to respond with none and this way we could keep the same trait bounds and `fn respond()` signature, if that makes sense.

Reply:
I kinda prefer the `into_inner` variant, for its explicitness. Just adding `respond` makes it a faux-interface like the containers in the rust stdlib, which I do not think is ideal. But I can be convinced otherwise :)

The `respond_none`/`respond_empty` is a bit tempting, but ultimately it makes it a little harder to understand, in my mind.

What do you guys think of renaming the newstruct, so it is not called "Responder"? Maybe `Autoclosing`?

Reply:
I just don't understand why regular `Responder` can have `respond` but this one can't?

Reply:
I updated as per the majority opinion in [d397288](https://github.com/casper-network/casper-node/pull/3035/commits/d3972884433d962c74480ea4b470587f880e5c19).